### PR TITLE
[FIX] l10n_es_edi_facutrae: fix TotalGrossAmount and new fields

### PR DESCRIPTION
**Pull request by alialfie**
[FIX] l10n_es_edi_facutrae: fix TotalGrossAmount and new fields

- The tag TotalGrossAmount should be the sum of the GrossAmount tag
  which is calculated as (price - discount + charge).
  Currently, the TotalGrossAmount is calculated as the sum of the price
  before subtracting the discount and adding the charge.
- Add new fields to the XML.
- Remove fields for total general discounts and charges as they were
  computed incorrectly.

task-4001821

### CODE REVIEW

File: addons/l10n_es_edi_facturae/models/account_move.py [hunk 1]
```diff
@@ -234,7 +234,7 @@ def _l10n_es_edi_facturae_inv_lines_to_items(self, conversion_rate=None):
             price_before_discount = sum(to_update['price_subtotal'] for _dummy, to_update in tax_before_discount['base_lines_to_update'])
             discount = max(0., (price_before_discount - line.price_subtotal))
             surcharge = abs(min(0., (price_before_discount - line.price_subtotal)))
-            totals['total_gross_amount'] += price_before_discount
+            totals['total_gross_amount'] += line.price_subtotal
             totals['total_general_discounts'] += discount
             totals['total_general_surcharges'] += surcharge
```
Comment (jco-odoo) [inline addons/l10n_es_edi_facturae/models/account_move.py:239]:
```suggestion
```

Comment (alialfie) [reply addons/l10n_es_edi_facturae/models/account_move.py:239]: Me and CHKL thought it's a good idea to only fix the issues the clients are currently talking about right now, and fix the general discounts/charges in another commit as they also need to be computed correctly. But for now we might as well remove those two lines. We also discussed more yesterday and there will be more fixes in this commit.

Comment (jco-odoo) [reply addons/l10n_es_edi_facturae/models/account_move.py:239]: @chklop @alialfie Ok, it is linked.

File: addons/l10n_es_edi_facturae/models/account_move.py [hunk 2]
```diff
@@ -254,7 +252,7 @@ def _l10n_es_edi_facturae_inv_lines_to_items(self, conversion_rate=None):
             totals['total_taxes_withheld'] += sum((abs(tax["tax_amount"]) for tax in taxes_withheld_computed))
 
             invoice_line_values.update({
-                'FileReference': self._l10n_es_edi_facturae_get_filename().split('.')[0][:20],
+                'FileReference': self.ref[:20] if self.ref else False,
```
Comment (jco-odoo) [inline addons/l10n_es_edi_facturae/models/account_move.py:255]: The change from name to ref is quite intrusive...

Comment (alialfie) [reply addons/l10n_es_edi_facturae/models/account_move.py:255]: In what way? It was incorrect in the first place so it has to be changed :sweat_smile:

**Comment by jco-odoo** [comment]
@robodoo r+

**Comment by alialfie** [comment]
@robodoo r+

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**